### PR TITLE
C# ResNet50 v2 sample/tutorial

### LINKED ISSUE #4709: Concrete C# samples
Whilst the [C# Inference Tutorial](https://github.com/microsoft/onnxruntime/blob/master/docs/CSharp_API.md) demonstrates the C# API surface, the C# samples are lacking and could be complimented with more concrete ones, like the [Python Samples](https://github.com/microsoft/onnxruntime/tree/master/samples#Python):

image

We could start with a C# version of the ResNet50 example and build from there. If in agreeance, I'd be happy to take this on, having just written [blog post](https://dev.to/marcusturewicz/serverless-image-classification-with-onnx-net-and-azure-functions-1h7l) about doing image classification on Azure Functions with the C# API. Let me know what you think.

### PR DESCRIPTION
**Pull request by marcusturewicz**
**Description**:
Adds a concrete C# sample/tutorial

**Motivation and Context**
Fixes #4709

### CODE REVIEW

File: csharp/sample/Microsoft.ML.OnnxRuntime.ResNet50v2Sample/Program.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+            // Setup inputs
+            var inputs = new List<NamedOnnxValue>
+            {
+                NamedOnnxValue.CreateFromTensor("data", input)
+            };
+
+            // Run inference
+            using var session = new InferenceSession(modelFilePath);
+            using IDisposableReadOnlyCollection<DisposableNamedOnnxValue> results = session.Run(inputs);
```
Comment (yuslepukhin) [inline csharp/sample/Microsoft.ML.OnnxRuntime.ResNet50v2Sample/Program.cs:59]: using(var session = new InferenceSession(mdoelFilePath))
using(var results = session.Run(inputs))
{
  // Put the rest of the function here so both session and results are still alive until the end of the scope.

}

File: csharp/sample/Microsoft.ML.OnnxRuntime.ResNet50v2Sample/README.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
+# C# Sample: ResNet50 v2
+
```
Comment (skottmckay) [inline csharp/sample/Microsoft.ML.OnnxRuntime.ResNet50v2Sample/README.md:2]: Thank you for such a clear and detailed readme!

Comment (marcusturewicz) [reply csharp/sample/Microsoft.ML.OnnxRuntime.ResNet50v2Sample/README.md:2]: No worries! 😊

**Comment by marcusturewicz** [comment]
CI failures look unrelated..?

**Pull request merged**